### DISCUSSION TITLE
Why is only the derivative term in a PID controller low-pass filtered, while the proportional and integral terms do not require low-pass filtering?

### DISCUSSION THREAD

**Discussion by HelloIDTC** [bitcraze/discussions#1536] [Q&A]
https://github.com/bitcraze/crazyflie-firmware/blob/master/src/utils/src/pid.c#L82

**Comment by gemenerik** [comment]
Low-pass filtering of only the derivative term is often done because differentiation amplifies high-frequency noise, which can lead to unstable and erratic control behavior. The filter helps attenuate these noise components, allowing the derivative action to focus on meaningful changes in the process, thereby improving overall system stability. As you can see in the code, whether or not only the derivative term is low-pass filtered depends on the used setting.
